bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/handler/LivisiBridgeHandler.java

```diff
@@ -0,0 +1,1004 @@
+        synchronized (this.lock) {
+            if (event.isLinkedtoDevice()) {
+                final boolean isSHCClassic = bridgeDevice.isClassicController();
+                bridgeDevice.getDeviceState().getState().getOperationStatus(isSHCClassic)
+                        .setValue(event.getProperties().getOperationStatus(isSHCClassic));
+                bridgeDevice.getDeviceState().getState().getCpuUsage(isSHCClassic)
+                        .setValue(event.getProperties().getCpuUsage(isSHCClassic));
+                bridgeDevice.getDeviceState().getState().getDiskUsage().setValue(event.getProperties().getDiskUsage());
+                bridgeDevice.getDeviceState().getState().getMemoryUsage(isSHCClassic)
+                        .setValue(event.getProperties().getMemoryUsage(isSHCClassic));
+                onDeviceStateChanged(bridgeDevice);
+            }
+        }
+    }
+
+    @Override
+    public void onEvent(final String msg) {
+        logger.trace("onEvent called. Msg: {}", msg);
+
+        try {
+            final Optional<EventDTO> eventOptional = parseEvent(msg);
+            if (eventOptional.isPresent()) {
+                EventDTO event = eventOptional.get();
+                switch (event.getType()) {
+                    case BaseEventDTO.TYPE_STATE_CHANGED:
+                    case BaseEventDTO.TYPE_BUTTON_PRESSED:
+                        handleStateChangedEvent(event, deviceStructMan);
+                        break;
+                    case BaseEventDTO.TYPE_DISCONNECT:
+                        logger.debug("Websocket disconnected.");
+                        scheduleRestartClient(true);
+                        break;
+                    case BaseEventDTO.TYPE_CONFIGURATION_CHANGED:
+                        handleConfigurationChangedEvent(event);
+                        break;
+                    case BaseEventDTO.TYPE_CONTROLLER_CONNECTIVITY_CHANGED:
+                        handleControllerConnectivityChangedEvent(event, deviceStructMan);
+                        break;
+                    case BaseEventDTO.TYPE_NEW_MESSAGE_RECEIVED:
+                    case BaseEventDTO.TYPE_MESSAGE_CREATED:
+                        final Optional<MessageEventDTO> messageEvent = gson.fromJson(msg, MessageEventDTO.class);
+                        if (messageEvent.isPresent()) {
+                            handleNewMessageReceivedEvent(Objects.requireNonNull(messageEvent.get()), deviceStructMan);
+                        }
+                        break;
+                    case BaseEventDTO.TYPE_MESSAGE_DELETED:
+                        handleMessageDeletedEvent(event, deviceStructMan);
+                        break;
+                    default:
+                        logger.debug("Unsupported event type {}.", event.getType());
+                        break;
+                }
+            }
+        } catch (IOException | ApiException | AuthenticationException | RuntimeException e) {
```

Comment:
I don't know for sure but I think this could at least occur when an invalid websocket event is received / can't get parsed successfully. In this case gson would throw a RuntimeException. The catch block handles this with handleClientException(...) which switches the bridge to offline and reloads everything.

Reply:
OK, let's leave it then.